src/remote.c

```diff
@@ -804,7 +851,8 @@ int git_remote__get_http_proxy(git_remote *remote, bool use_ssl, char **proxy_ur
 	}
 
 	/* HTTP_PROXY / HTTPS_PROXY environment variables */
-	error = git__getenv(&val, use_ssl ? "HTTPS_PROXY" : "HTTP_PROXY");
+	git_buf_init(&buf, 0);
```

Comment:
The buffer has already been initialised. If what you wanted was to clear its contents, use `git_buf_clear()`.